pmlc/dialect/stdx/ir/ops.cc

```diff
@@ -91,6 +91,80 @@ LogicalResult verifyAtomicRMWYieldOp(AtomicRMWYieldOp op) {
   return success();
 }
 
+// ---- FPToSIOp ----
+
+void printFPToSIOp(OpAsmPrinter &p, FPToSIOp op) {
+  p << op.getOperation()->getName() << ' ' << op.getOperand() << " : "
+    << op.getOperand().getType() << " to "
+    << op.getOperation()->getResult(0).getType();
+}
+
+ParseResult parseFPToSIOp(OpAsmParser &parser, OperationState &result) {
+  OpAsmParser::OperandType srcInfo;
+  Type srcType, dstType;
+  return failure(parser.parseOperand(srcInfo) ||
+                 parser.parseOptionalAttrDict(result.attributes) ||
+                 parser.parseColonType(srcType) ||
+                 parser.resolveOperand(srcInfo, srcType, result.operands) ||
+                 parser.parseKeywordType("to", dstType) ||
+                 parser.addTypeToList(dstType, result.types));
+}
+
+LogicalResult verifyFPToSIOp(FPToSIOp op) {
```

Comment:
I think we can do this verification declaratively with ODS. In other words, if we tweak the acceptable types for the `arguments` and `results` in the `.td` file, then this will generate logically equivalent C++ verify code.